File: src/main/java/vergisst/minecraftmod/weaponthrow/handler/ConfigRegistry.java
package vergisst.minecraftmod.weaponthrow.handler;

import me.shedaniel.autoconfig.AutoConfig;
import me.shedaniel.autoconfig.ConfigHolder;
import me.shedaniel.autoconfig.serializer.GsonConfigSerializer;
import vergisst.minecraftmod.weaponthrow.config.WeaponThrowConfig;

public class ConfigRegistry {

    public static ConfigHolder<WeaponThrowConfig> COMMON;

    public static void registerConfig() {
        COMMON = AutoConfig.register(WeaponThrowConfig.class, GsonConfigSerializer::new);
    }
}

File: src/main/kotlin/vergisst/minecraftmod/weaponthrowlite/handlers/ConfigRegistry.kt
package vergisst.minecraftmod.weaponthrowlite.handlers

import me.shedaniel.autoconfig.AutoConfig
import me.shedaniel.autoconfig.ConfigHolder
import me.shedaniel.autoconfig.serializer.GsonConfigSerializer

import vergisst.minecraftmod.weaponthrowlite.config.WeaponThrowConfig

object ConfigRegistry {
    lateinit var COMMON: ConfigHolder<WeaponThrowConfig>

    fun registerConfig() {
        COMMON = AutoConfig.register(
            WeaponThrowConfig::class.java,
//            ::GsonConfigSerializer
        ) { config, clazz ->
            GsonConfigSerializer(config, clazz)
        }
    }
}